### ISSUE TITLE
Retry fetches that fail via LoadFromFile over http (applying non-LoadFromFile routing)

### ISSUE THREAD

**Issue by oschaaf** [pagespeed/ngx_pagespeed#952]
It might be nice if LoadFromFile would have an extra (optional) argument (bool), that would indicate that the fetch should be retried if it couldn't be found on the filesystem. 
The retried fetch would then disregard this LoadFromFile configuration but instead follow the rest of the configured routing scheme,.
Being able to retry would also allow a single an easier setup when to-be-optimised content is a mix of static, virtual resources, and overlapping MapProxyDomain lines.

**Comment by jeffkaufman** [comment]
This worries me, because it only covers *some* of the ways `LoadFromFile` can bite people.  If you have a WSGI app on `/wsgi` running with `uwsgi` then your proposal will work great: the files corresponding to `/wsgi/some/path` won't be found on the filesystem and so will be loaded from `uwsgi` as they should be.  But many scripting languages use in-place scripts sitting under the `www` hierarchy, like php.  If your php setup has `/foo.php` then you do expect that file to exist on disk and you don't want to send it out uninterpreted in response to requests.

In many setups what you propose would be easier, allowing people to just turn on `LoadFromFile` globally, but it would be really hard to document when that was safe and when it would send uninterpreted or access controlled files out.

**Comment by oschaaf** [comment]
@jeffkaufman I agree. Adding to that - I've seen more then a handful of instances where `LoadFromFile` was just configured to work from the root - without further restriction. The docs seem fine to me, so my thought is that it's not a problem over there. 

Thinking out loud: perhaps LoadFromFile should come with a whitelist of extensions it applies to out of the box?

**Comment by jeffkaufman** [comment]
In many cases an extension whitelist would let people just turn this on at the server level, which would be pretty cool.  LoadFromFile can only handle extensions it knows about anyway, because it needs to know what to set for the content type.
